internal/api/commitment.go

```diff
@@ -1668,3 +2118,123 @@ func (p *v1Provider) UpdateCommitmentDuration(w http.ResponseWriter, r *http.Req
 
 	respondwith.JSON(w, http.StatusOK, map[string]any{"commitment": c})
 }
+
+// DelegateCommitmentCheck decides whether LiquidClient.ChangeCommitments() should be called,
+// depending on the setting of liquid.ResourceInfo.HandlesCommitments. If not, it routes the
+// operation to be performed locally on the database. In case the LiquidConnection is not filled,
+// a LiquidClient is instantiated on the fly to perform the operation. It utilizes a given ServiceInfo so that no
+// double retrieval is necessary caused by operations to assemble the liquid.CommitmentChange.
+func (p *v1Provider) DelegateChangeCommitments(ctx context.Context, req liquid.CommitmentChangeRequest, serviceType db.ServiceType, serviceInfo liquid.ServiceInfo, dbi db.Interface) (result liquid.CommitmentChangeResponse, err error) {
+	localCommitmentChanges := liquid.CommitmentChangeRequest{
+		DryRun:      req.DryRun,
+		AZ:          req.AZ,
+		InfoVersion: req.InfoVersion,
+		ByProject:   make(map[liquid.ProjectUUID]liquid.ProjectCommitmentChangeset),
+	}
+	remoteCommitmentChanges := liquid.CommitmentChangeRequest{
+		DryRun:      req.DryRun,
+		AZ:          req.AZ,
+		InfoVersion: req.InfoVersion,
+		ByProject:   make(map[liquid.ProjectUUID]liquid.ProjectCommitmentChangeset),
+	}
+	for projectUUID, projectCommitmentChangeset := range req.ByProject {
+		for resourceName, resourceCommitmentChangeset := range projectCommitmentChangeset.ByResource {
+			// this is just to make the tests deterministic because time.Local != local IANA time (after parsing)
+			for i, commitment := range resourceCommitmentChangeset.Commitments {
+				commitment.ExpiresAt = commitment.ExpiresAt.Local()
+				confirmBy, exists := commitment.ConfirmBy.Unpack()
+				if exists {
+					commitment.ConfirmBy = Some(confirmBy.Local())
+				}
```

Comment:
```suggestion
				commitment.ConfirmBy = options.Map(commitment.ConfirmBy, time.Time.Local)
```
I recently learned that you can refer to a method that way.